tools/version-checker/update-version

```diff
@@ -3,10 +3,30 @@
 const fs = require('fs');
 const fetch = require('node-fetch');
 
-const getLatestVersion = async (currentVersion) => { 
-    const res = await fetch(`https://web.whatsapp.com/check-update?version=${currentVersion}&platform=web`);
-    const data = await res.json();
-    return data.currentVersion;
+const getLatestVersion = async () => { 
+    const response = await fetch('https://web.whatsapp.com/', {
+        headers: {
+            'user-agent':
+                'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/129.0.0.0 Safari/537.36',
+            'accept-language': 'en-US,en;q=1',
+            'sec-fetch-mode': 'navigate',
+            'cache-control': 'no-cache',
+            cookie: 'wa_lang_pref=en;wa_build=c',
+            pragma: 'no-cache',
+        },
+    });
+
+    const text = await response.text();
+    
+    if (text) {
+        const matches = text.match(/"client_revision"\s*:\s*(\d+)/) || [];
+
+        if (matches[1]) {
+            return `2.3000.${matches[1]}`;
```

Comment:
Why is `2.3000.` always prepended? Can this value be retrieved from anywhere else?

Reply:
@pedroslopez
Yes, for sure.
Before executing the script, we install all dependencies anyway ([here](https://github.com/pedroslopez/whatsapp-web.js/blob/1e03ca682850240872028034e51a2727cde1c98e/.github/workflows/update.yml#L21)), so we can use a puppeteer to retrieve the WWeb version.